src/multi_pieces.rs

```diff
@@ -92,24 +92,15 @@ where
     }
 }
 
-impl<C> HeapSize for FMIndexMultiPiecesBackend<C, ()>
+impl<C, S> HeapSize for FMIndexMultiPiecesBackend<C, S>
 where
-    C: Character,
-{
-    fn heap_size(&self) -> usize {
-        self.bw.heap_size() + self.cs.capacity() * std::mem::size_of::<u64>()
-    }
-}
-
-impl<C> HeapSize for FMIndexMultiPiecesBackend<C, SOSampledSuffixArray>
-where
-    C: Character,
+    S: HeapSize,
 {
     fn heap_size(&self) -> usize {
         self.bw.heap_size()
-            + self.cs.capacity() * std::mem::size_of::<u64>()
-            + self.suffix_array.size()
-            + self.doc.capacity() * std::mem::size_of::<usize>()
+            + self.cs.len() * std::mem::size_of::<u64>()
```

Comment:
Ah, `cs` a `Vec<usize>` now - I'll rewrite `size_of::<u64>()` to `size_of::<usize>()` in this PR.

Reply:
Fixed https://github.com/ajalab/fm-index/pull/94/commits/6aa3bab5a62dfa389cb4026da899397a4cfa7df8